src/d3a/d3a_core/sim_results/endpoint_buffer.py

```diff
@@ -159,10 +159,8 @@ def update_area_aggregated_stats(self, area):
 
     def _update_bids_offer_trades(self, area):
         if area.current_market is not None:
-            self.bids_offers_trades[area.uuid] = area.current_market.get_bids_offers_trades()
-            # TODO: (D3ASIM-2670) Trades are already stored in bids_offers_trades:
-            #  accumulation of area.stats.market_trades endpoint should be deprecated
-            self.last_energy_trades_high_resolution[area.uuid] = area.stats.market_trades
+            self.bids_offers_trades[area.uuid] = \
+                {area.current_market.time_slot_str: area.current_market.get_bids_offers_trades()}
```

Comment:
I agree in general with this implementation, but I am not sure you really need the time_slot_str of the current market here. The current market time is already transmitted in a separate field in the simulation results, therefore you can populate it from there. This would save you the effort in stripping away this time slot in the d3a-web PR (essentially instead of sending it and removing it, you could not send it at all here and add it via the results.current_market)

In addition to that, is this format absolutely required for the UI implementation of `energy_trades_high_resolution` ? If not I would prefer to not send it at all, since these are either way reported per requested market. cc'ing @hannesdiedrich  to pitch in

Reply:
I agree. AFAIS, you can omit calling `normalized_bid_offer_trade_data` on the d3a-web side then.